Emby.Server.Implementations/Library/Validators/CollectionValidator.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using MediaBrowser.Controller.Collections;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Entities.Movies;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Model.Querying;
+using Microsoft.Extensions.Logging;
+using Jellyfin.Data.Enums;
+
+namespace Emby.Server.Implementations.Library.Validators
+{
+    /// <summary>
+    /// Class CollectionValidator.
+    /// </summary>
+    public class CollectionValidator
+    {
+        /// <summary>
+        /// The library manager.
+        /// </summary>
+        private readonly ILibraryManager _libraryManager;
+
+        /// <summary>
+        /// The collection manager.
+        /// </summary>
+        private readonly ICollectionManager _collectionManager;
+
+        /// <summary>
+        /// The logger.
+        /// </summary>
+        private readonly ILogger<CollectionValidator> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CollectionValidator" /> class.
+        /// </summary>
+        /// <param name="libraryManager">The library manager.</param>
+        /// <param name="collectionManager">The collection manager.</param>
+        /// <param name="logger">The logger.</param>
+        public CollectionValidator(ILibraryManager libraryManager, ICollectionManager collectionManager, ILogger<CollectionValidator> logger)
+        {
+            _libraryManager = libraryManager;
+            _collectionManager = collectionManager;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Runs the specified progress.
+        /// </summary>
+        /// <param name="progress">The progress.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>Task.</returns>
+        public async Task Run(IProgress<double> progress, CancellationToken cancellationToken)
+        {
+            var movies = _libraryManager.GetItemList(new InternalItemsQuery
+            {
+                IncludeItemTypes = new[] { nameof(Movie) },
+                IsVirtualItem = false,
+                OrderBy = new List<ValueTuple<string, SortOrder>>
+                {
+                new ValueTuple<string, SortOrder>(ItemSortBy.SortName, SortOrder.Ascending)
+                },
+                Recursive = true
+            }).Select(m => m as Movie).ToList();
```

Comment:
Doing the cast and then calling ToList allocates a new list of the same objects. You already know they are `Movie` types, so just do the cast in the for-loop.